cocos/core/data/decorators/utils.ts

```diff
@@ -58,7 +58,7 @@ export const emptyDecorator: ClassDecorator & LegacyPropertyDecorator = () => {}
  * @zh
  * 一个忽略所有参数并且返回 `emptyDecorator` 的函数。
  */
-export const emptyDecoratorFn = () => emptyDecorator;
+export const emptyDecoratorFn = (): ClassDecorator & LegacyPropertyDecorator => emptyDecorator;
```

Comment:
The return type is strange, please double check it.

Reply:
It's OK, `ClassDecorator & LegacyPropertyDecorator` is the type of `emptyDecorator` object

this is an arrow function, and `emptyDecorator` is the return value, not part of the return type

Reply:
Why use `&`, i don't understand.

Reply:
`&` means emptyDecorator need to implement the both ClassDecorator and LegacyPropertyDecorator

Reply:
But why not `|`?